lib/widgets/wn_animated_avatar.dart

```diff
@@ -0,0 +1,175 @@
+      animationController.repeat();
+
+      final imageProvider = NetworkImage(pictureUrl!);
+      final imageStream = imageProvider.resolve(ImageConfiguration.empty);
+
+      void onImage(ImageInfo info, bool synchronousCall) {
+        currentLoadedUrl.value = pictureUrl;
+        previousLoadedUrl.value = null;
+        isLoading.value = false;
+        animationController.stop();
+        animationController.reset();
+      }
+
+      void onError(Object error, StackTrace? stackTrace) {
+        isLoading.value = false;
+        animationController.stop();
+        animationController.reset();
+      }
+
+      final listener = ImageStreamListener(onImage, onError: onError);
+      imageStream.addListener(listener);
+
+      return () => imageStream.removeListener(listener);
+    }, [pictureUrl]);
+
+    final hasCurrentImage = currentLoadedUrl.value == pictureUrl;
+    final showPreviousImage = isLoading.value && previousLoadedUrl.value != null;
+
+    if (pictureUrl == null || pictureUrl!.isEmpty) {
+      return WnInitialsAvatar(
+        displayName: displayName,
+        size: avatarSize,
+      );
+    }
+
+    return Container(
+      width: avatarSize,
+      height: avatarSize,
+      constraints: BoxConstraints.tight(Size(avatarSize, avatarSize)),
+      decoration: BoxDecoration(
+        shape: BoxShape.circle,
+        color: colors.backgroundSecondary.withValues(alpha: 0.4),
+        border: Border.all(color: colors.foregroundTertiary, width: 1.5),
+      ),
+      clipBehavior: Clip.antiAlias,
+      child: Stack(
+        fit: StackFit.expand,
+        children: [
+          if (!isLoading.value || showPreviousImage)
+            _InitialsContent(
+              displayName: displayName,
+              size: avatarSize,
+            ),
+
+          if (showPreviousImage)
+            Image.network(
+              previousLoadedUrl.value!,
+              width: avatarSize,
+              height: avatarSize,
+              fit: BoxFit.cover,
+            ),
+
+          if (hasCurrentImage)
+            Image.network(
+              currentLoadedUrl.value!,
+              width: avatarSize,
+              height: avatarSize,
+              fit: BoxFit.cover,
+            ),
+
+          if (isLoading.value)
+            AnimatedBuilder(
+              animation: animationController,
+              builder: (context, _) {
+                final opacity = 1.0 - (animationController.value / 2);
+                return Opacity(
+                  opacity: opacity,
+                  child: WnAnimatedPixelOverlay(
+                    progress: animationController.value,
+                    color: colors.backgroundPrimary,
+                    width: avatarSize,
+                    height: avatarSize,
+                    pixelSize: 4.0.w,
+                  ),
+                );
+              },
+            ),
+        ],
+      ),
+    );
+  }
+}
+
+class _InitialsContent extends StatelessWidget {
```

Comment:
🦥 : I find confusing that there already is an initials avatar but here the logic is repeated... because of borders an decoration?